.github/workflows/ci.yaml

```diff
@@ -2,9 +2,14 @@ name: ci
 
 on:
   push:
-    branches: ['*']
+    branches:
+      # only on merges to master branch
+      - master
+      # and version branches, which only include minor versions (eg: v3.4)
+      - v[0-9]+.[0-9]+
     tags:
-      - v*
+      # also version tags, which include bugfix releases (eg: v3.4.0)
+      - v[0-9]+.[0-9]+.[0-9]+
   pull_request:
     type: [opened, reopened, edited]
```

Comment:
Technically we could / should probably also filter pull requests so builds are only triggered if it's a PR against master branch (and perhaps a version one?).